raiden/utils/filters.py

```diff
@@ -123,40 +132,33 @@ def _do_get_new_entries(self, from_block: BlockSpecification, to_block: BlockSpe
             to_block=to_block,
         )
         result = self.web3.eth.getLogs(filter_params)
-        self._last_block = to_block
+        self._last_block = block_specification_to_number(block=to_block, web3=self.web3)
         return result
 
-    def get_new_entries(self, target_block_number: BlockSpecification):
-        latest_block_number = self.web3.eth.blockNumber
+    def get_new_entries(self, target_block_number: BlockNumber) -> List[Dict[str, Any]]:
         with self._lock:
-            result = []
-            from_block = max(
-                self.filter_params.get('fromBlock', GENESIS_BLOCK_NUMBER),
-                self._last_block + 1,
+            result: List[Dict[str, Any]] = []
+            filter_from_number = block_specification_to_number(
+                block=self.filter_params.get('fromBlock', GENESIS_BLOCK_NUMBER),
+                web3=self.web3,
             )
-            to_block = self.filter_params.get('toBlock')
-            if to_block in (None, 'latest', 'pending'):
-                if not isinstance(target_block_number, int):
-                    target_block_number = latest_block_number
-
-            else:
-                target_block_number = to_block
+            from_block_number = max(filter_from_number, self._last_block + 1)
```

Comment:
Removed the old extra logic for choices between `toBlock` and `target_block_number` and sticking to target block number now. It made no sense and is not well documented so I will see if tests pass to determine whether it was un-necessary.

Reply:
Tests passed. I assume then the change is fine.